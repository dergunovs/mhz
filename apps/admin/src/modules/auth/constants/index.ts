export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
