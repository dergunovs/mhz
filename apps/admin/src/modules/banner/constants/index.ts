export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
