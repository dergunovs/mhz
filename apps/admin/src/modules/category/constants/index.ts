export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
