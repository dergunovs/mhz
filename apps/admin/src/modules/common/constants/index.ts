export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
