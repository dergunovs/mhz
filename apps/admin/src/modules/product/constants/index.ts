export const URL_PRODUCT = '/products';
export const URL_PRODUCT_CREATE = `${URL_PRODUCT}/create`;
export const URL_PRODUCT_EDIT = `${URL_PRODUCT}/edit`;
