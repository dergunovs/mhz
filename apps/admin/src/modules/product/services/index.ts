import { Ref, ComputedRef } from 'vue';

import { useQuery, useMutation, IPageQuery } from 'mhz-helpers';
import { API_PRODUCT } from 'mhz-contracts';

import {
  getProductsApi,
  getProductApi,
  postProductApi,
  updateProductApi,
  deleteProductApi,
} from '@/product/services/api';

export function getProducts(query: Ref<IPageQuery | number>) {
  return useQuery({ queryKey: [API_PRODUCT, query], queryFn: () => getProductsApi(query) });
}

export function getProduct(id?: ComputedRef<string | string[]>) {
  return useQuery({ queryKey: [API_PRODUCT, id], queryFn: () => getProductApi(id) });
}

export function postProduct(options: object) {
  return useMutation({ mutationKey: [API_PRODUCT], mutationFn: postProductApi, ...options });
}

export function updateProduct(options: object) {
  return useMutation({ mutationKey: [API_PRODUCT], mutationFn: updateProductApi, ...options });
}

export function deleteProduct(options: object) {
  return useMutation({ mutationKey: [API_PRODUCT], mutationFn: deleteProductApi, ...options });
}
