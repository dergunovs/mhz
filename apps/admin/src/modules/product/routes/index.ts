import { URL_PRODUCT } from '@/product/constants';

export const productRoutes = [
  {
    path: URL_PRODUCT,
    name: 'ProductList',
    component: () => import('@/product/pages/ProductListPage.vue'),
  },
  {
    path: `${URL_PRODUCT}/:id`,
    name: 'Product',
    component: () => import('@/product/pages/ProductPage.vue'),
  },
];
