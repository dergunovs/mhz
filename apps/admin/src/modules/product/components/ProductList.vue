<template>
  <UiTable :headers="tableHeaders">
    <template v-if="props.categories?.length">
      <tr v-for="product in props.categories" :key="product._id">
        <td data-grow>
          <RouterLink :to="`${URL_PRODUCT_EDIT}/${product._id}`">
            {{ product.title }}
          </RouterLink>
        </td>
        <td data-no-wrap>
          {{ formatDateTime(product.date_created) }}
        </td>
        <td data-no-wrap>
          {{ formatDateTime(product.date_updated) }}
        </td>
        <td>
          <UiButton @click="mutate(product._id)" layout="plain">Delete</UiButton>
        </td>
      </tr>
    </template>
  </UiTable>
</template>

<script setup lang="ts">
import { useQueryClient } from '@tanstack/vue-query';

import { IProduct } from 'mhz-types';
import { UiTable, UiButton, toast } from 'mhz-ui';
import { formatDateTime } from 'mhz-helpers';

import { deleteProduct } from '@/product/services';
import { API_PRODUCT, URL_PRODUCT_EDIT } from '@/product/constants';

interface IProps {
  categories?: IProduct[];
}

const props = defineProps<IProps>();

const tableHeaders = ['Product', 'Created', 'Updated', ''];

const queryClient = useQueryClient();

const { mutate } = deleteProduct({
  onSuccess: async () => {
    await queryClient.refetchQueries({ queryKey: [API_PRODUCT], exact: true });
    toast.success('Product deleted');
  },
});
</script>
