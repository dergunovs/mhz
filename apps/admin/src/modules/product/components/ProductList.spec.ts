import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import ProductList from './ProductList.vue';

import { wrapperFactory } from '@/common/test';
import { URL_PRODUCT_EDIT } from '@/product/constants';
import { PRODUCTS } from '@/product/fixtures';

const productListRow = dataTest('product-list-row');
const productListLink = dataTest('product-list-link');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(ProductList, {
    props: {
      products: PRODUCTS.data,
    },
  });
});

enableAutoUnmount(afterEach);

describe('ProductList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(ProductList)).toBeTruthy();
  });

  it('shows rows if products props', async () => {
    expect(wrapper.findAll(productListRow).length).toEqual(PRODUCTS.data.length);

    await wrapper.setProps({ products: [] });

    expect(wrapper.find(productListRow).exists()).toBe(false);
  });

  it('generated link to product', async () => {
    const LINK = `${URL_PRODUCT_EDIT}/${PRODUCTS.data[0]._id}`;

    expect(wrapper.find(productListLink).attributes('to')).toEqual(LINK);
  });

  it('shows product content in cells', async () => {
    expect(wrapper.find(productListLink).text()).toEqual(PRODUCTS.data[0].title);
  });
});
