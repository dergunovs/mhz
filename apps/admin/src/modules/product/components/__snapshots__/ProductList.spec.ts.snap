// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductList > matches snapshot 1`] = `
"<table headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" isloading="false" lang="en">
  <tbody>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df8a7d7ac901ef3e273343" data-test="product-list-link">G.Skill Trident Z5 Neo RGB 2x32 GB DDR5 6000</a></td>
      <td>RAM</td>
      <td data-no-wrap="">G.Skill</td>
      <td data-no-wrap="">31200 ₽</td>
      <td>Yes</td>
      <td>24</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">21 дек. 2023 г.</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df89e67ac901ef3e2732d8" data-test="product-list-link">G.Skill Flare X5 2x16 GB DDR5 5600</a></td>
      <td>RAM</td>
      <td data-no-wrap="">G.Skill</td>
      <td data-no-wrap="">12200 ₽</td>
      <td>Yes</td>
      <td>3</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df89a87ac901ef3e273298" data-test="product-list-link">G.Skill TRIDENT Z5 2x16 GB DDR5 5600</a></td>
      <td>RAM</td>
      <td data-no-wrap="">G.Skill</td>
      <td data-no-wrap="">11700 ₽</td>
      <td>Yes</td>
      <td>2</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df89747ac901ef3e27327a" data-test="product-list-link">AMD Radeon R9 Gamer Series 1x8 GB DDR4 3200</a></td>
      <td>RAM</td>
      <td data-no-wrap="">AMD</td>
      <td data-no-wrap="">1700 ₽</td>
      <td>Yes</td>
      <td>4</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">0</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df893f7ac901ef3e27325c" data-test="product-list-link">AMD Radeon R9 Gamer Series 1x16 GB DDR4 3200</a></td>
      <td>RAM</td>
      <td data-no-wrap="">AMD</td>
      <td data-no-wrap="">3300 ₽</td>
      <td>Yes</td>
      <td>3</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">0</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df88e97ac901ef3e273209" data-test="product-list-link">AMD Radeon R9 Gamer Series 2x8 GB DDR4 3600</a></td>
      <td>RAM</td>
      <td data-no-wrap="">AMD</td>
      <td data-no-wrap="">3800 ₽</td>
      <td>Yes</td>
      <td>9</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df88a07ac901ef3e2731dd" data-test="product-list-link">AMD Radeon R9 Gamer Series 2x16 GB DDR4 3200</a></td>
      <td>RAM</td>
      <td data-no-wrap="">AMD</td>
      <td data-no-wrap="">6600 ₽</td>
      <td>Yes</td>
      <td></td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">0</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df883a7ac901ef3e2731d0" data-test="product-list-link">Samsung M378A4G43AB2 1x32 GB DDR4 3200</a></td>
      <td>RAM</td>
      <td data-no-wrap="">Samsung</td>
      <td data-no-wrap="">7200 ₽</td>
      <td>Yes</td>
      <td>3</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df87f77ac901ef3e2731ba" data-test="product-list-link">Samsung M378A2K43EB1 1x16 GB DDR4 3200</a></td>
      <td>RAM</td>
      <td data-no-wrap="">Samsung</td>
      <td data-no-wrap="">3800 ₽</td>
      <td>Yes</td>
      <td></td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df87c27ac901ef3e2731b2" data-test="product-list-link">Samsung M378A1G44AB0 1x8 GB DDR4 3200</a></td>
      <td>RAM</td>
      <td data-no-wrap="">Samsung</td>
      <td data-no-wrap="">2650 ₽</td>
      <td>Yes</td>
      <td>3</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">0</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df86477ac901ef3e2731a6" data-test="product-list-link">Thermaltake H-ONE 2x8 GB DDR4 3200</a></td>
      <td>RAM</td>
      <td data-no-wrap="">Thermaltake</td>
      <td data-no-wrap="">4400 ₽</td>
      <td>Yes</td>
      <td>6</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">0</td>
    </tr>
    <tr data-test="product-list-row">
      <td data-grow=""><a to="/products/edit/64df860a7ac901ef3e27319c" data-test="product-list-link">Thermaltake TOUGHRAM 2x8 GB DDR4 4400</a></td>
      <td>RAM</td>
      <td data-no-wrap="">Thermaltake</td>
      <td data-no-wrap="">7700 ₽</td>
      <td>Yes</td>
      <td>7</td>
      <td data-no-wrap="">18 авг. 2023 г.</td>
      <td data-no-wrap="">0</td>
    </tr>
  </tbody>
</table>"
`;
